File: uniapp_flutter/lib/utils/project_submission_upload_status.dart
enum ProjectSubmissionUploadStatus {
  UNSYNCED,
  SYNCED,
  SYNCED_WITH_MEDIA,
  VALIDATION_ERROR,
  SERVER_ERROR,
  FAILED,
  DELETED,
  APP_VERSION_MISMATCH_ERROR
}
class ProjectSubmissionUploadStatusHelper {
  static int getValue(ProjectSubmissionUploadStatus status){
    switch(status){
      case ProjectSubmissionUploadStatus.UNSYNCED:
        return 0;
      case ProjectSubmissionUploadStatus.SYNCED:
        return 1;
      case ProjectSubmissionUploadStatus.SYNCED_WITH_MEDIA:
        return 2;
      case ProjectSubmissionUploadStatus.SERVER_ERROR:
        return -2;
      case ProjectSubmissionUploadStatus.FAILED:
        return -3;
      case ProjectSubmissionUploadStatus.DELETED:
        return -4;
      case ProjectSubmissionUploadStatus.APP_VERSION_MISMATCH_ERROR:
        return -5;
      default:
        return 5;
    }
  }
  static final Map<int, ProjectSubmissionUploadStatus> valueToStateNameMap = new Map();
  final int value = 0;
  static createMap() {

    for (ProjectSubmissionUploadStatus myEnum in ProjectSubmissionUploadStatus.values) {
      valueToStateNameMap[ProjectSubmissionUploadStatusHelper.getValue(myEnum)]= myEnum;
    }
  }

  static ProjectSubmissionUploadStatus getStateByValue(int value) {
    if(valueToStateNameMap.isEmpty) {
      createMap();
    }
    return valueToStateNameMap[value];
  }
}

File: app/app/src/main/java/com/vassar/unifiedapp/utils/ProjectSubmissionUploadStatus.java
package com.vassar.unifiedapp.utils;
import android.provider.ContactsContract;

import java.util.HashMap;
import java.util.Map;

public enum ProjectSubmissionUploadStatus {

    UNSYNCED(0),
    SYNCED(1),
    SYNCED_WITH_MEDIA(2),
    VALIDATION_ERROR(-1),
    SERVER_ERROR(-2),
    FAILED(-3),
    DELETED(-4),
    APP_VERSION_MISMATCH_ERROR(-5);

    private static final Map<Integer, ProjectSubmissionUploadStatus> valueToStateNameMap = new HashMap<>();
    private final Integer value;

    static {
        for (ProjectSubmissionUploadStatus myEnum : values()) {
            valueToStateNameMap.put(myEnum.getValue(), myEnum);
        }
    }

    private ProjectSubmissionUploadStatus(final Integer newValue) {
        value = newValue;
    }

    public Integer getValue() {
        return value;
    }

    public static ProjectSubmissionUploadStatus getStateByValue(Integer value) {
        return valueToStateNameMap.get(value);
    }
}
